do_with_in_base/src/lib.rs

```diff
@@ -2950,6 +2950,60 @@ pub fn runMarkersHandler<T: StartMarker + Clone>(c: Configuration<T>, v: Variabl
+/// 
+/// ## q mode ##
+/// 
+/// If the `array` token is followed immediately by a `q`, the command will treat its arguments as quoted and handle them as such.
+/// For example, `%(array q length %(quote {{1}}))` returns 1.
+/// 
+/// ## Commands ##
+/// 
+/// | Command     | Summary                                                                                   | Syntax |
+/// | -------     | -------                                                                                   | ------ |
+/// | `length`    | Return the length of the array.                                                           | `$(array length <array>)`                             |
+/// | `ith`       | Work with array using positional arguments.                                               | `$(array ith <operation> <arguments>`                 |
+/// | `slice`     | TODO                                                                                      |         |
+/// | `concat`    | Concatenate the elements of two or more arrays together.                                  | `$(array concat <array1> <array2> ...)`               |
+/// | `each`      | Passes array of argument tuples to a handler.                                             | `$(array each <handlerName> <array>)`                 |
+/// | `map`       | Returns a new array based on applying <block> to <srcArray>                               | `$(array map <should_isolate> <name> <block> <srcArray>)` |
+/// | `mk`        | Iterates through arguments, checks if they are valid array elements, and groups them      | `$(array mk <block1> <block2> ... )`                  |
+/// 
+/// Some of these commands do a lot:
+/// 
+/// ## `ith` ##
+/// 
+/// All `ith` subcommands take a zero-indexed position argument to indicate where in the array the subcommand applies.
+/// Negative integers may be used to index from the tail of the array e.g. `-1` represents the final element, `-2` the penultimate element, and so on.
+/// The special tokens `head` and `tail` can be used instead of integer positions 0 and -1 respectively.
+/// 
+/// | Subcommand  | Summary | Syntax |
+/// | ---------   | ------- | ------ |
+/// | `get`       | Returns the element at `<position>`.                          | `$(array ith get <position> $array);` |
+/// | `set`       | Replace the element at `<position>` with `<newEl>`.           | `$(array ith set <position> <newEl> $array);` |
+/// | `insert`    | Insert element `<newEl>` before the element at `<position>`.  | `$(array ith insert <position> <newEl> $array)` |
+/// | `remove`    | Remove the element at `<position>`.                           | `$(array ith remove <position> $array)` |
+/// 
+/// ## `each` ##
+/// 
+/// Iterates through each item in <array> and runs the handler identified by <handlerName>, passing the item's sub-elements as arguments.
+/// For example,
+///  `$(array each run [ {{let f = 3;}} {{let g = 4;}} ] )` will set `$f` to 3 and `$g` to 4.
```

Comment:
Example should perhaps be more like: `$(array each run [ {$(let f = {3})} {$(let g = {4})} ] )`